### PR TITLE
Implement utility methods for adding/removing vertices and edges of graphs; allow custom mobjects as vertices

### PR DESCRIPTION
**Pull request by behackl**
## Changelog / Overview

- Implement `Graph.add_vertices`, `Graph.remove_vertices`, `Graph.add_edges`, `Graph.remove_edges`; including custom animation overrides.
- Introduce new keyword argument `vertex_mobjects` for `Graph.__init__`, allowing to pass already existing mobjects that should be used as vertices.

## Motivation
Modifying existing Graphs is rather tedious at the moment, this PR tries to make interaction with a graph after its creation a bit easier.

## Testing Status
Added new tests for all the added functionality, tests pass locally.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/graph.py [hunk 1]
```diff
@@ -446,6 +460,469 @@ def __getitem__(self: "Graph", v: Hashable) -> "Mobject":
...
+        ----------
+
+        vertices
+            A list of vertices to be removed from the graph.
+
+        Examples
+        --------
+
+        >>> G = Graph([1, 2, 3], [(1, 2), (2, 3)])
+        >>> G.remove_vertices(2, 3)
```
Comment (Nikhil0504) [inline manim/mobject/graph.py:711]: Maybe try print the statement to give the desired output

File: manim/mobject/graph.py [hunk 2]
```diff
@@ -446,6 +460,469 @@ def __getitem__(self: "Graph", v: Hashable) -> "Mobject":
...
+        Group
+            A mobject containing all removed objects.
+
+        Examples
+        --------
+        ::
+
+            >>> G = Graph([1, 2, 3], [(1, 2), (2, 3)])
+            >>> G.remove_vertex(2)
+            VGroup(Line, Line, Dot)
```
Comment (Nikhil0504) [inline manim/mobject/graph.py:663]: Try using type()

**Review by Nikhil0504 — COMMENTED** [review]
Gave some feedback about the docs error

**Comment by Darylgolden** [comment]
At a first glance, LGTM! Since `add_vertex` and `add_vertices` (and all the corresponding add and remove methods) do the same thing, I think there should only be one user-facing method, `add_vertices` (and all the corresponding plural add/remove methods). The singular methods can be made private. This prevents essentially duplicating documentation.

**Review by Darylgolden — CHANGES_REQUESTED** [review]
At a first glance, LGTM! Since `add_vertex` and `add_vertices` (and all the corresponding add and remove methods) do the same thing, I think there should only be one user-facing method, `add_vertices` (and all the corresponding plural add/remove methods). The singular methods can be made private. This prevents essentially duplicating documentation.

**Comment by behackl** [comment]
> At a first glance, LGTM! Since `add_vertex` and `add_vertices` (and all the corresponding add and remove methods) do the same thing, I think there should only be one user-facing method, `add_vertices` (and all the corresponding plural add/remove methods). The singular methods can be made private. This prevents essentially duplicating documentation.

Thanks for looking at it! I took https://doc.sagemath.org/html/en/reference/graphs/sage/graphs/generic_graph.html as the inspiration for the interface. I could make the operations on one single object (`add_vertex`, `remove_vertex`, `add_edge`, `remove_edge`) private and remove the corresponding animation overrides; that would trim down the code required by a fair amount.

The only thing I do dislike in this case is that if only the plural-methods exist, the interface for operating on a single object would be weird -- the argument still has to be passed in a list in that case. Maybe the arguments should be changed to starred lists in that case.

**Comment by behackl** [comment]
> At a first glance, LGTM! Since `add_vertex` and `add_vertices` (and all the corresponding add and remove methods) do the same thing, I think there should only be one user-facing method, `add_vertices` (and all the corresponding plural add/remove methods). The singular methods can be made private. This prevents essentially duplicating documentation.

I've changed the interface according to your suggestion (and do like it more myself that way as well).

Given that I've written (comparatively) extensive tests for this feature, and that there aren't too many people usually reviewing graph code, I'd still like to merge this before the release. I'll not merge in case there are any objections whatsoever, just let me know. (No hard feelings either. :-))

**Comment by Darylgolden** [comment]
LGTM! Feel free to merge :)

**Pull request merged**